app/views/katello/sync_management/index.html.erb

```diff
@@ -1,17 +1,19 @@
-<%= javascript 'katello/sync_management' %>
+<% content_for(:javascripts) do -%>
```

Comment:
```suggestion
<% content_for(:katello_javascripts) do -%>
```

THIS fixes it.

the reset happens here:
https://github.com/Katello/katello/blob/136a73e8c007d4ed5969adfb6411dc2c68e9880d/app/views/katello/layouts/katello.html.erb#L9-L13

Reply:
```
<% content_for(:katello_javascripts) do -%>
  <%= javascript_include_tag 'katello/sync_management' %>
  <script>
    localize({
      "cancel": "<%= escape_javascript(_('Cancel')) %>",
      "cancelled": "<%= escape_javascript(_('Sync Canceled')) %>",
      "error": "<%= escape_javascript(_('Error')) %>",
      "complete": "<%= escape_javascript(_('Sync complete.')) %>",
      "no_start_time": "<%= escape_javascript(_('No start time currently available.')) %>"
    });

      KT.repo_status = $.parseJSON('<%= escape_javascript(@repo_status.to_json.html_safe) %>');
      KT.permissions = { "syncable" : <%= any_syncable? %> };
  </script>
<% end -%>
```

this is how it looks in full on my box now

Reply:
I still get "Uncaught TypeError: Cannot read properties of undefined (reading 'select_repo')" on select none with this?

Reply:
Works for me (on a prod install, don't have a devel katello right now)

Reply:
This is the code generated for loading (I added line breaks for readability):
```
<script type="text/javascript">
      const html = "
        <script src=\"/assets/katello/common/vendor-9f3c77c20de190d174986979ad7d0d5cee9362a9a280adabf643e57ee4259155.js\"><\/script>\n
        <script src=\"/assets/katello/common-41006bb366e1429ef7397de66336e76035be453bc9bac600cd845b9571266232.js\"><\/script>\n
        <script src=\"/assets/katello/sync_management-db168612191a6dc850c8d5cd28375ee6af25d8fd011e62638f11f7fd903f5025.js\"><\/script>\n
        <script>\n    localize({\n      \"cancel\": \"Cancel\",\n      \"cancelled\": \"Sync Canceled\",\n      \"error\": \"Error\",\n      \"complete\": \"Sync complete.\",\n      \"no_start_time\": \"No start time currently available.\"\n    });\n\n      KT.repo_status = \$.parseJSON(\'{\\\"1\\\":{\\\"id\\\":1,\\\"product_id\\\":1,\\\"progress\\\":{},\\\"state\\\":\\\"Never Synced\\\",\\\"raw_state\\\":\\\"never_synced\\\"}}\');\n      KT.permissions = { \"syncable\" : true };\n  <\/script>\n\n"
      load_dynamic_javascripts(html);
    </script>
```

Reply:
I put it in the wrong file :facepalm: 
but setting `var KT = KT ? KT : {};` instead of `var KT = {};`  in `app/assets/javascripts/katello/common/katello.global.js` also fixed it :upside_down_face:

Reply:
Yeah… but that still means that the code gets executed twice, which seems wrong :)

Reply:
I mean… It's probably still a good idea to have such a safety guard, but it should not be the "real" fix

Reply:
which code gets executed twice?

Reply:
`katello/common` gets executed twice if we use `content_for(:javascripts)` (and not `:katello_javascripts`), as that generates:
- katello/common (from sync_management/index.html)
- katello/sync_management
- katello/common (from layouts/katello)

Reply:
> I mean… It's probably still a good idea to have such a safety guard, but it should not be the "real" fix

Yeah, I thought you meant adding it will cause the double execution.
Added it to the pr and used your solution, thanks!